ibm_watson/websocket/recognize_listener.py

```diff
@@ -196,16 +196,20 @@ def on_data(self, ws, message, message_type, fin):
         elif 'results' in json_object or 'speaker_labels' in json_object:
             hypothesis = ''
             if 'results' in json_object:
-                hypothesis = json_object['results'][0]['alternatives'][0][
-                    'transcript']
-                b_final = (json_object['results'][0]['final'] is True)
-                transcripts = self.extract_transcripts(
-                    json_object['results'][0]['alternatives'])
+                if len(json_object['results']) != 0:
```

Comment:
Wondering if we can do something `if 'results' in json_object and if len(json_object['results']) != 0` instead.

Reply:
It does really seems a cleaner and more robust approach. I'd like to emphasize that the changes I proposed were the same changes we needed in our project to make it work, any propositions and edits from this repo maintainers will be happily accepted.